core/src/main/java/io/onetable/delta/DeltaDataFileUpdatesExtractor.java

```diff
@@ -61,37 +59,54 @@ public class DeltaDataFileUpdatesExtractor {
 
   public Seq<Action> applySnapshot(
       DeltaLog deltaLog, List<OneFileGroup> partitionedDataFiles, OneSchema tableSchema) {
-    List<OneDataFile> currentDataFiles = new ArrayList<>();
-    try (DataFileIterator fileIterator =
-        deltaDataFileExtractor.iteratorWithoutStats(deltaLog.snapshot(), tableSchema)) {
-      fileIterator.forEachRemaining(currentDataFiles::add);
-      OneDataFilesDiff filesDiff =
-          OneDataFilesDiff.from(
```

Comment:
Do we want to do something similar for Iceberg and Hudi targets? Is there any way we can make this logic common?

Reply:
That's a good point. 
This logic could be applied to other clients as well. I am going to create a new issue to make this logic more general and implement it for other clients. Merging this PR in the meantime should unblock the team who is waiting for this fix. I will also mark the new issue as a good beginner issue in case any new contributor is interested.